### PR TITLE
[libmupdf] Update mupdf to 1.21.1

### PR DESCRIPTION
**Pull request by simon987**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

The `use-if-instead-of-ifdef-in-writer.patch` was added by me in https://github.com/microsoft/vcpkg/pull/20328, but it has since been committed upstream in https://github.com/ArtifexSoftware/mupdf/commit/305adc250bffaf so I removed it.

### CODE REVIEW

File: ports/libmupdf/portfile.cmake [hunk 1]
```diff
@@ -3,12 +3,11 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ArtifexSoftware/mupdf
-    REF 61b63d734a7b9df618f6b45dda2466aed442f7f0 # 1.19.0-rc2
-    SHA512 16661c012e18ac72b24c46caf5c02515c29a05e0a8dcf95076eff3a1f2e87c225245037480ed37068858fe6e04ff4a404f69877599b208ab9265d054ec117820
+    REF f39eee48564274635e231e2ab652221cce2ec1bd # 1.21.1
+    SHA512 aa16d0939739f96c66ee802032c321adcebc86c9fa5bfe7ec64f20614377a4f61c358270216b57a1e4b1929337c4cc856d9ec4666ed8f0edaeaa7f2336a84ac3
```
Comment (Cheney-W) [inline ports/libmupdf/portfile.cmake:7]:
```suggestion
    REF "${VERSION}"
    SHA512 756bdc6911fe64c5e07d4c5c7e5e9497e384c2c9d60d6b9edd6f419723c69719e25accd7a999bdd8093691fd5d91821d8ba7a35b9a1ce02f6394c3ca96e62f53
```

File: ports/libmupdf/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libmupdf",
-  "version": "1.19.0-rc2",
-  "port-version": 1,
+  "version": "1.21.1",
+  "port-version": 2,
```
Comment (Cheney-W) [inline ports/libmupdf/vcpkg.json:4]:
```suggestion
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Comment by simon987** [comment]
@Cheney-W I made the suggested changes.

However I think I screwed up the version thing because of that force-push, any idea on how to resolve this?

**Comment by Cheney-W** [comment]
```
vcpkg x-add-version libmupdf --overwrite-version
git add .
git commit -m "update json"
git push
```

**Comment by Cheney-W** [comment]
Pinging @simon987  for response. Is work still being done for this PR?

**Comment by Cheney-W** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by simon987** [comment]
A newer version is available so I will create a new PR instead

**Pull request closed without merging**